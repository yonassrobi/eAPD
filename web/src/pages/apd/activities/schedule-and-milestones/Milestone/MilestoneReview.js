import PropTypes from 'prop-types';
import React, { useMemo } from 'react';
import Review from '../../../../../components/Review';
import { stateDateToDisplay } from '../../../../../util';

const MilestoneReview = ({
  index,
  item: { endDate, milestone },
  expand,
  onDeleteClick
}) => {
  const formattedDate = useMemo(() => stateDateToDisplay(endDate), [endDate]);

  return (
    <Review
      heading={`${index + 1}. ${milestone || 'Milestone not specified'}`}
      headingLevel="4"
      onDeleteClick={onDeleteClick}
      onEditClick={expand}
      ariaLabel={`${index + 1}. ${milestone || 'Milestone not specified'}`}
      objType="Milestone"
    >
      <p className="ds-u-margin-top--2">
        <strong>Target completion date:</strong> {formattedDate}
      </p>
    </Review>
  );
};

MilestoneReview.propTypes = {
  expand: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  item: PropTypes.shape({
    endDate: PropTypes.string.isRequired,
    milestone: PropTypes.string.isRequired
  }).isRequired,
  onDeleteClick: PropTypes.func
};

MilestoneReview.defaultProps = {
  onDeleteClick: null
};

export default MilestoneReview;
