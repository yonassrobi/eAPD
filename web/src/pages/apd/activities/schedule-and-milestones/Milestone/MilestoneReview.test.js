import { shallow } from 'enzyme';
import React from 'react';

import MilestoneReview from './MilestoneReview';

describe('the MilestoneReview component', () => {
  const expand = jest.fn();
  const onDelete = jest.fn();

  const component = shallow(
    <MilestoneReview
      index={1532}
      item={{
        // The Battle of Britain ends with the Nazis unable to gain air
        // superiority to support an invasion. The invasion is postponed until
        // the following year, but in fact never happens.
        endDate: '1940-10-31',
        milestone: 'Milestone name'
      }}
      expand={expand}
      onDeleteClick={onDelete}
    />
  );

  test('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  test('triggers the delete event', () => {
    expect(component.find('Review').prop('onDeleteClick')).toEqual(onDelete);
  });

  test('hooks up the expand event', () => {
    expect(component.find('Review').prop('onEditClick')).toEqual(expand);
  });
});
