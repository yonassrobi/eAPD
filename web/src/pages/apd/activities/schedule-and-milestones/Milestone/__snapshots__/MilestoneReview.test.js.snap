// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the MilestoneReview component renders correctly 1`] = `
<Review
  ariaLabel="1533. Milestone name"
  editHref={null}
  heading="1533. Milestone name"
  headingLevel="4"
  objType="Milestone"
  onDeleteClick={[MockFunction]}
  onDeleteLabel={null}
  onEditClick={[MockFunction]}
  skipConfirmation={false}
  title={null}
>
  <p
    className="ds-u-margin-top--2"
  >
    <strong>
      Target completion date:
    </strong>
     
    10/31/1940
  </p>
</Review>
`;
